bundles/org.eclipse.equinox.p2.discovery/src/org/eclipse/equinox/internal/p2/discovery/Catalog.java

```diff
@@ -73,10 +73,10 @@ public IStatus performDiscovery(IProgressMonitor monitor) {
 
 		final int totalTicks = 100000;
 		final int discoveryTicks = totalTicks - (totalTicks / 10);
-		monitor.beginTask(Messages.Catalog_task_discovering_connectors, totalTicks);
+		SubMonitor subMonitor = SubMonitor.convert(monitor, Messages.Catalog_task_discovering_connectors, totalTicks);
 		try {
 			for (AbstractDiscoveryStrategy discoveryStrategy : discoveryStrategies) {
-				if (monitor.isCanceled()) {
```

Comment:
Is this extra check necessary? We do a split (which auto checks for cancellations ) just a few lines later

Reply:
> Is this extra check necessary? We do a split (which auto checks for cancellations ) just a few lines later

This is not completely the same. In one case you can silently return in the other an exception is thrown.

Reply:
Split also throws this exception

Reply:
That's true. But given that the method is supposed to return gracefully upon cancellation, rather than throwing an exception, I should use `newChild()` rather than `split()`.